refactor(ivy): remove unused embedded view instructions

AndrewKushnir opened the pull request:
This commit performs a cleanup and removes unused embedded view instructions and corresponding tests.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

AndrewKushnir committed:
refactor(core): remove unused embedded view instructions

This commit performs a cleanup and removes unused embedded view instructions and corresponding tests.

pkozlowski-opensource commented on packages/core/src/render3/interfaces/container.ts:
```diff
@@ -48,9 +48,7 @@ export const CONTAINER_HEADER_OFFSET = 9;
- *  - Transplanted `LView`s (see: `LView[DECLARATION_COMPONENT_VIEW])`
+ * Used to track Transplanted `LView`s (see: `LView[DECLARATION_COMPONENT_VIEW])`
  */
 export const enum ActiveIndexFlag {
```
This flag / field should be probably renamed / cleaned up (for example, do we still need `INCREMENT`? Put simply, this field was originally having JS block embedded view index and was later re-purposed to track info about transplanted views. But with JS blocks gone we should make the intention behind the `ACTIVE_INDEX` more clear?

AndrewKushnir replied:
I did a quick, search and it looks like `INCREMENT` is the only unused field at this moment, so I removed it. I believe the use of other fields might also be refactored and they can be removed, but I think that might be the next step in refactoring - to revisit existing logic and data structures to clean them up once this PR lands.

pkozlowski-opensource commented on packages/core/test/render3/query_spec.ts:
```diff
@@ -1367,66 +1366,6 @@ describe('query', () => {
     });
   });
 
-  describe('queryList', () => {
```
It would be good to re-write those as integration / acceptance tests using ngIf / ngFor

pkozlowski-opensource commented on packages/core/test/render3/providers_spec.ts:
```diff
@@ -1316,99 +1146,6 @@ describe('providers', () => {
     });
   });
-
-  describe('lifecycles', () => {
```
probably would make sense to move it to a `TestBed` acceptance test (if we don't have any similar one).

pkozlowski-opensource commented on packages/core/test/render3/providers_spec.ts:
```diff
@@ -859,176 +859,6 @@ describe('providers', () => {
-              '<div><repeated>foo1,2</repeated><repeated>foo1,2</repeated><repeated>foo1,2</repeated></div>');
-    });
-
-    it('should have access to viewProviders of the repeated component', () => {
```
probably would make sense to move it to a `TestBed` acceptance test (if we don't have any similar one).

pkozlowski-opensource commented on packages/core/test/render3/providers_spec.ts:
```diff
@@ -859,176 +859,6 @@ describe('providers', () => {
   });
 
-  describe('- embedded views', () => {
-    it('should have access to viewProviders of the host component', () => {
```
probably would make sense to move it to a `TestBed` acceptance test (if we don't have any similar one).

pkozlowski-opensource commented:
@AndrewKushnir thnx for starting this cleanup - soooooo many things will be easier when this cleanup lands! 

I did the first pass on this PR and left some small comment. More importantly, though, I think that we should migrate more of the tests from the `test/render3` folder to `TestBed`. I've left some comments / suggestions but IMO we should migrate all tests that:
* are not specifically testing for DOM insertion with JS blocks;
* don't exist as an acceptance test already. 

I know that it will be more effort but would be great not to loose test coverage....

AndrewKushnir commented:
Hi @pkozlowski-opensource, thanks for the review. I've refactored and moved the tests that you mentioned to the `acceptance` folder. Could you please let me know if there are some other tests that would benefit from moving to acceptance folder? Thank you.

pkozlowski-opensource commented on packages/core/test/render3/renderer_factory_spec.ts:
```diff
@@ -104,102 +104,4 @@ describe('renderer factory lifecycle', () => {
-    expect(ngDevMode).toHaveProperties({rendererDestroy: 0, rendererDestroyNode: 3});
-  });
-
-  it('should call renderer.destroy for each component destroyed', () => {
```
@AndrewKushnir I would expect that we've got already an acceptance test like this but if not I would see it as valuable to migrate.

AndrewKushnir replied:
Thanks @pkozlowski-opensource. I moved these tests to `acceptance/renderer_factory_spec.ts`, so we have the necessary coverage.

pkozlowski-opensource reviewed (approved):
This LGTM overall, just left one comment regarding a test that we might want to migrate. Other tests seem to be either JS-blocks specific or exercising very basic scenarios (which, I would assume, are already covered in other acceptance tests).

AndrewKushnir commented:
[Presubmit](http://test/OCL:291081011:BASE:291081036:1579749317262:c12a275d)

AndrewKushnir commented:
[Presubmit](http://test/OCL:291485328:BASE:291485339:1579923712184:67d42e1b)

mhevery reviewed (approved):
I saw a lot of Green, but I am assuming that most of it is you just moving things around so I did not look very deeply into it.

AndrewKushnir committed:
refactor(core): rename `instructions/container.ts` -> `instructions/template.ts`

Prior to this change, the `template` instruction logic was located in the `instructions/container.ts` file alongside embedded view instructions. Since unused embedded view instructions are removed in a previous commit, this commit renames `container.ts` -> `template.ts`, since only template-related instructions were retained.

kara commented on packages/core/test/render3/listeners_spec.ts:
```diff
@@ -287,277 +287,6 @@ describe('event listeners', () => {
     expect(ctx.showing).toBe(false);
   });
 
-  it('should support listeners in views', () => {
```
Seems like this test could be moved to acceptance/listener_spec.ts

kara commented on packages/core/test/render3/listeners_spec.ts:
```diff
@@ -287,277 +287,6 @@ describe('event listeners', () => {
-    expect(comp.counter).toEqual(2);
-  });
-
-  it('should destroy listeners in for loops', () => {
```
Seems like this test could be moved to acceptance/listener_spec.ts

kara commented on packages/core/test/render3/listeners_spec.ts:
```diff
@@ -718,254 +447,6 @@ describe('event listeners', () => {
     expect(comp.counter).toEqual(6);
   });
 
-  it('should destroy listeners in nested views', () => {
```
Seems like this test could be moved to acceptance/listener_spec.ts

kara reviewed (approved):
LGTM, just need to rebase and add the requested tests

Reviewed-for: global-approvers

AndrewKushnir committed:
fixup! refactor(core): remove unused embedded view instructions

AndrewKushnir commented:
Thanks for review @kara 👍 I've added requested tests in a fixup commit https://github.com/angular/angular/pull/34715/commits/9d09e376c897abf6f4b6a9f3712b4c65c26460da.

AndrewKushnir commented:
[Presubmit](https://test.corp.google.com/ui#id=OCL:310475215:BASE:310475287:1588900898928:bd222a5b)
